File: sdk/samples/WFSamples/Applications/OutlookWorkflowWizard/CS/ReadEmailActivity/BaseMailbox.cs
//---------------------------------------------------------------------
//  This file is part of the Windows Workflow Foundation SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//---------------------------------------------------------------------

using System;
using System.ComponentModel;
using System.Workflow.ComponentModel;

namespace Microsoft.Samples.Workflow.OutlookWorkflowWizard
{
    public partial class BaseMailbox: Activity
    {
        public BaseMailbox()
        {
            InitializeComponent();
        }

        private FilterOption filter;
        public FilterOption Filter
        {
            get { return filter; }
            set { filter = value; }
        }

        private String filterValue;
        public string FilterValue
        {
            get { return filterValue; }
            set { filterValue = value; }
        }
    }

    public enum FilterOption
    {
        [Description("Subject")]
        Subject = 0,
        [Description("From")]
        From = 1,
        [Description("To")]
        To = 2,
        [Description("CC")]
        CC = 3,
        [Description("BCC")]
        Bcc = 4,
    }
}

File: sdk/samples/WFSamples/Applications/OutlookWorkflowWizard/VB/ReadEmailActivity/BaseMailbox.vb
'---------------------------------------------------------------------
'  This file is part of the Windows Workflow Foundation SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'---------------------------------------------------------------------

Imports System
Imports System.ComponentModel
Imports System.Workflow.ComponentModel

Public Class BaseMailbox
    Inherits Activity

    Dim filterOptionValue As FilterOption
    Public Property Filter() As FilterOption
        Get
            Return Me.filterOptionValue
        End Get
        Set(ByVal value As FilterOption)
            Me.filterOptionValue = value
        End Set
    End Property

    Dim filterValueValue As String
    Public Property FilterValue() As String
        Get
            Return Me.filterValueValue
        End Get
        Set(ByVal value As String)
            Me.filterValueValue = value
        End Set
    End Property
End Class

Public Enum FilterOption
    <Description("Subject")> _
    Subject = 0
    <Description("From")> _
    FromEmail = 1
    <Description("To")> _
    [To] = 2
    <Description("CC")> _
    CC = 3
    <Description("BCC")> _
    Bcc = 4
End Enum